src/plots/array_container_defaults.js

```diff
@@ -45,17 +49,15 @@ module.exports = function handleArrayContainerDefaults(parentObjIn, parentObjOut
 
     for(var i = 0; i < contIn.length; i++) {
         var itemIn = contIn[i],
-            itemOut = {};
+            itemOut = {},
+            itemOpts = {};
 
         if(!Lib.isPlainObject(itemIn)) {
-            opts.itemIsNotPlainObject = true;
+            itemOpts.itemIsNotPlainObject = true;
             itemIn = {};
         }
-        else {
-            opts.itemIsNotPlainObject = false;
-        }
 
-        opts.handleItemDefaults(itemIn, itemOut, parentObjOut, opts);
+        opts.handleItemDefaults(itemIn, itemOut, parentObjOut, opts, itemOpts);
```

Comment:
Sure, that's fine. I still don't quite see where the handler will use the original `opts` but I guess it doesn't hurt to leave it in.